packages/datetime/src/dateRangePicker.tsx

```diff
@@ -356,7 +374,15 @@ export class DateRangePicker extends AbstractComponent<IDateRangePickerProps, ID
         />
     );
 
-    private handleDayMouseEnter = (day: Date, modifiers: DayPicker.DayModifiers) => {
+    private handleDayMouseEnter = (
+        day: Date,
+        modifiers: ReactDayPicker.DayModifiers,
+        e: React.MouseEvent<HTMLDivElement>,
+    ) => {
+        if (typeof this.props.dayPickerProps.onDayMouseEnter === "function") {
```

Comment:
Again, use `CoreUtils.safeInvoke`:

```tsx
CoreUtils.safeInvoke(this.props.dayPickerProps.onDayMouseEnter, day, modifiers, e);
```

I keep going back and forth about where we should be invoking this within the function. e.g. if the day is `disabled`, should we still invoke the consumer's custom callback? @giladgray